Fixed timestep indexing error in Langevin dyanmics plotting.

burichh opened the pull request:
**Bug**: Currently the plotter does not plot at the specified timesteps `[0, 334, 668, 999]`, but at the first four timesteps `[0, 1, 2, 3]`. This is due to the fact that the `t_idx` is used as the time index at `xt = xts[:, t_idx]`.

**Fix**: Use the `indices_to_plot[t_idx]` to index out the points: `xt = xts[:, indices_to_plot[t_idx]]`.

**Note**: I had to decrease `sigma` from `sigma=10.0` to  `sigma=0.6`, otherwise the Langevin process pulls the Gaussian distribution to the target GaussianMixutre distribution way too fast, and all the points defined by `plot_every=334` will be already very closely fitting to the target distribution (and I guess your intention was to show the continuous transition from source Gaussian distribution to target GaussianMixture distribution via the Langevin process).

With `sigma=10.0`:

![image](https://github.com/user-attachments/assets/6c070418-e8ff-4481-8b4c-20a4156919f2)

With `sigma=0.6`

![image](https://github.com/user-attachments/assets/0188cfb9-473a-4f27-8ccd-475ca288f103)

eje24 commented:
Hi @burichh, thanks for opening this PR. I would recommend several additional changes before we merge:
1. Address the comment I added about using `plot_xts`. 
2. Similarly update `animate_dynamics` by adding an `animate_xts = xts[:, indices_to_animate]` [here](https://github.com/eje24/iap-diffusion-labs/blob/main/labs/lab_one.ipynb?short_path=b6471ea#L991).
3. Add these updates to the solutions as well. 

Thanks!

burichh commented:
> Hi @burichh, thanks for opening this PR! I would recommend several additional changes before we merge:
> 
> 1. Address the comment I added about using `plot_xts`.
> 2. Similarly update `animate_dynamics` by adding an `animate_xts = xts[:, indices_to_animate]` [here](https://github.com/eje24/iap-diffusion-labs/blob/main/labs/lab_one.ipynb?short_path=b6471ea#L991).
> 3. Add these updates to the solutions as well.
> 
> Thanks!

Hi Ezra,

Thanks for the feedback, I added the modifications you suggested. Additionally, in the animation cells I increased the time range from [0.0, 1.5] to [0.0, 5.0] to match the range used in the plotting cells.

Please have a look. Thanks!

eje24 commented:
Hi @burichh , looks great. Merging now. Thanks!
